metacat-thrift/src/main/java/com/netflix/metacat/thrift/CatalogThriftHiveMetastore.java

```diff
@@ -1115,7 +1156,21 @@ public Partition get_partition_with_auth(
      */
     @Override
     public PartitionsByExprResult get_partitions_by_expr(final PartitionsByExprRequest req) throws TException {
-        throw unimplemented("get_partitions_by_expr", new Object[]{req});
+        return requestWrapper("get_partitions_by_expr", new Object[]{req},
+            () -> {
+                String filter = null;
+                if (req.getExpr() != null) {
+                    filter = Utilities.deserializeExpressionFromKryo(req.getExpr()).getExprString();
+                }
+                if (req.getExpr() == null || filter != null) {
+                    //TODO: We need to handle the case for 'hasUnknownPartitions'
+                    final List<Partition> partitions =
+                        getPartitionsByFilter(req.getDbName(), req.getTblName(), filter, req.getMaxParts());
+                    return new PartitionsByExprResult(partitions, false);
+                } else {
+                    throw new MetaException("Failed to deserialize expression - ExprNodeDesc not present");
+                }
+            });
```

Comment:
These set of conditions are very confusing. If the exception is from the ```Utilities.deserializeExpressionFromKryo```, it should be after the first if () like
```
if (req.getExpr() != null) {
           filter = Utilities.deserializeExpressionFromKryo(req.getExpr()).getExprString();
          if ( filter == null ) {
                 throw new MetaException("Failed to deserialize expression - ExprNodeDesc not present");
          }
}
```
Then the second if is the else of if (req.getExpr() != null)
```
if (req.getExpr() != null) {
xxxx
}
else {
                    final List<Partition> partitions =
                      getPartitionsByFilter(req.getDbName(), req.getTblName(), filter, req.getMaxParts());
                    return new PartitionsByExprResult(partitions, false);
}
```

Reply:
Done.